lib/datasource/remote/api/invoice_repository.dart

```diff
@@ -2,22 +2,23 @@ import 'package:async/async.dart';
 
 // ignore: import_of_legacy_library_into_null_safe
 import 'package:dart_esr/dart_esr.dart' as esr;
+import 'package:seeds/datasource/local/models/token_data_model.dart';
 import 'package:seeds/datasource/remote/api/eos_repository.dart';
 import 'package:seeds/datasource/remote/firebase/firebase_remote_config.dart';
 import 'package:seeds/datasource/remote/model/token_model.dart';
 import 'package:seeds/domain-shared/app_constants.dart';
 
 class InvoiceRepository extends EosRepository {
   Future<Result<dynamic>> createInvoice(
-      {required double quantity, required String accountName, required TokenModel token, String? memo}) {
```

Comment:
I dont think the change from quantity to TokenDataModel is needed. 

This repository is better of not knowing anything about our Domain data models `TokenDataModel`

`tokenAmount.asFormattedString()` can be passed here instead of the entire model.

Reply:
and while at it. pls refactor TokenModel out as well and just pass in the `token.contract` from the user case.

Reply:
Makes sense, thanks!

Reply:
Actually - the reason we use an amountDataModel here instead of quantity is the question

"quantity of _what_?"

A quantity is meaningless without defining what the quantity is of. 

So tokenDataModel more generally answers this question and IMO should be used anywhere we pass around quantities.  

If I give you 1... then you'll be asking... 1 what? 1 dollar? 1 Seeds? 1 house? 

Code-wise why the quantity of what is important here, is because that's how we can format the quantity for the transaction. 

For that, it needs to know the symbol and precision - both encapsulated in TokenDataModel

I agree on the token model, I removed it and replaced with a String

If I replace quantity with the double that it was, then I need to either
- pass in precision and symbol as well - which is already in TokenDataModel
or
- pass the formatted string, which is a more loose and error prone definition of the data in TokenDataModel. 

So on second thought TokenDataModel should be in there, that's exactly what it is for - it has all the info we need, quantity, symbol, and precision, and nothing we don't need.

Reply:
And this is when you create a request object. And pass that object into the repo. 

Ill not stop this PR for this, since we dont enforce this in all places. 

But the repo should not know about the domain data layer. 

"pass in precision and symbol as well - which is already in TokenDataModel" - Nothing wrong with this. Or create a request object thats passed into the repo.

Reply:
Hmmm... a request object - are we doing that anywhere else @gguijarro-c-chwy ?

Reply:
I dont think so. Which is why i said i wont block this PR. 

One of my next tasks is to improve our models. 

Separate data models and domain models 

So, im cool with this for now. Ill fix it later